src/appleseed/foundation/image/filteredtile.cpp

```diff
@@ -179,14 +179,19 @@ float FilteredTile::compute_weighted_pixel_variance(
     const float rcp_second_weight = second_weight == 0.0f ? 0.0f : 1.0f / second_weight;
 
     // Get colors and assign weights.
-    Color4f main_color(abs(main[0]), abs(main[1]), abs(main[2]), abs(main[3]));
+    Color4f main_color(main[0], main[1], main[2], main[3]);
     main_color *= rcp_main_weight;
 
-    Color4f second_color(abs(second[0]), abs(second[1]), abs(second[2]), abs(second[3]));
+    Color4f second_color(second[0], second[1], second[2], second[3]);
     second_color *= rcp_second_weight;
 
+    const float rgb = abs(main_color.r + main_color.g + main_color.b);
```

Comment:
You just changed the semantics and I'm not sure this is correct anymore. What do you _really_ want to do?

Reply:
Negative values occurs because of filtering, but we can't compute `sqrt` of invalid value and we can't just ignore those values.

Reply:
That's not what I meant. Just before my review you were adding absolute values, now you're taking the absolute value of the sum.

Reply:
oh yes it's not the same, sorry :(

Reply:
updated.